File: agents/tests/test_purpose_guide.py
import unittest

from agents.purpose_guide_agent import PurposeGuide


class MockProfileDB:
    def __init__(self):
        self.called_with = None

    def get_profile(self, user_id):
        self.called_with = user_id
        return {"user_id": user_id, "name": "Test User", "dob": "2000-01-01"}


class MockZKProofTool:
    def __init__(self):
        self.called_with = None

    def generate_over18_proof(self, user_id):
        self.called_with = user_id
        return {"proof": "FAKE_PROOF", "publicSignals": {"is_over_18": 1}}


class MockWalletTool:
    def __init__(self):
        self.called_with = None

    def mint_sbt(self, user_id, proof):
        self.called_with = (user_id, proof)
        return "0xFAKE_TX_HASH"


class PurposeGuideTest(unittest.TestCase):
    def test_handle_mint_flow(self):
        profile_db = MockProfileDB()
        zk_tool = MockZKProofTool()
        wallet = MockWalletTool()

        agent = PurposeGuide(profile_db, zk_tool, wallet)
        result = agent.handle_mint('user-123')

        # Verify return structure
        self.assertIsInstance(result, dict)
        self.assertEqual(result.get('status'), 'submitted')
        self.assertIn('tx_hash', result)

        # Verify mocks were called with expected args
        self.assertEqual(profile_db.called_with, 'user-123')
        self.assertEqual(zk_tool.called_with, 'user-123')
        self.assertEqual(wallet.called_with[0], 'user-123')
        self.assertEqual(wallet.called_with[1]['proof'], 'FAKE_PROOF')


if __name__ == '__main__':
    unittest.main()

File: agents/tests/test_purpose_guide.js
const assert = require('assert');
const PurposeGuide = require('../purpose_guide_agent');

class MockProfileDB {
  constructor() { this.calledWith = null; }
  getProfile(userId) { this.calledWith = userId; return { user_id: userId, name: 'Test User', dob: '2000-01-01' }; }
}

class MockZKProofTool {
  constructor() { this.calledWith = null; }
  generateOver18Proof(userId) { this.calledWith = userId; return { proof: 'FAKE_PROOF', publicSignals: { is_over_18: 1 } }; }
}

class MockWalletTool {
  constructor() { this.calledWith = null; }
  mintSbt(userId, proof) { this.calledWith = [userId, proof]; return '0xFAKE_TX_HASH'; }
}

(function runTests(){
  const profileDb = new MockProfileDB();
  const zkTool = new MockZKProofTool();
  const wallet = new MockWalletTool();

  const agent = new PurposeGuide(profileDb, zkTool, wallet);
  agent.handleMint('user-123').then(result => {
    try {
      assert.strictEqual(result.status, 'submitted');
      assert.ok(result.tx_hash);

      assert.strictEqual(profileDb.calledWith, 'user-123');
      assert.strictEqual(zkTool.calledWith, 'user-123');
      assert.strictEqual(wallet.calledWith[0], 'user-123');
      assert.strictEqual(wallet.calledWith[1].proof, 'FAKE_PROOF');

      console.log('PurposeGuide JS tests passed');
      process.exit(0);
    } catch (err) {
      console.error('Test failed:', err.message);
      process.exit(2);
    }
  }).catch(err => { console.error('Test run error:', err); process.exit(3); });
})();
